graphql_compiler/compiler/expressions.py

```diff
@@ -394,10 +394,19 @@ def to_cypher(self):
                              u'should not be called.')
 
     def to_sql(self, aliases, current_alias):
-        """Not implemented, should not be used."""
-        raise AssertionError(u'GlobalContextField is not used as part of the query emission '
-                             u'process in SQL, so this is a bug. This function '
-                             u'should not be called.')
+        """Return a sqlalchemy Column picked from the appropriate alias."""
+        self.validate()
+        if isinstance(self.field_type, GraphQLList):
+            raise NotImplementedError(u'The SQL backend does not support lists. Cannot '
+                                      u'process field {}.'.format(self.location.field))
+
+        if self.location.field is not None:
+            if '@' in self.location.field:
+                raise NotImplementedError(u'The SQL backend does not support __typename.')
```

Comment:
We don't conclusively know that this is `__typename` here, all we know is that the field is a special field of some sort (there could be more than one in the future).

That said, this is because of an OrientDB-ism (`@class`), so maybe open an issue about using `__typename` as the field name in IR locations and refer to it from a comment here?

Reply:
Addressing this in [#628 ](https://github.com/kensho-technologies/graphql-compiler/pull/628)